src/lib/loaders/loaders_with_authentication/vortex.ts

```diff
@@ -34,15 +36,98 @@ export default (accessToken, opts) => {
     { method: "POST" }
   )
 
+  const vortexGraphqlLoader = ({ query, variables }: LoaderArgs) =>
+    vortexLoader(
+      "/graphql",
+      { query, variables: JSON.stringify(variables) },
+      {
+        method: "POST",
+      }
+    )
+
   return {
     vortexTokenLoader,
-    vortexGraphqlLoader: ({ query, variables }: LoaderArgs) =>
-      vortexLoader(
-        "/graphql",
-        { query, variables: JSON.stringify(variables) },
-        {
-          method: "POST",
+    vortexGraphqlLoader,
+    marketPriceInsightsBatchLoader: async (
+      params: { artistId: string; medium: string; category: string }[]
+    ) => {
+      // TODO: Fix this logic once we only need category to fetch insights
+      const artistIDMediumTuples = params.map((artist) => ({
+        artistId: artist.artistId,
+        medium: getVortexMedium(artist.medium, artist.category),
+      }))
+
+      try {
+        const vortexResult = await vortexGraphqlLoader({
+          query: gql`
+        query MarketPriceInsightsBatchQuery($artistIDMediumTuples: [ArtistIdMediumTupleType!]!) {
+          marketPriceInsightsBatch(input: $artistIDMediumTuples, first: ${MAX_MARKET_PRICE_INSIGHTS}) {
+            totalCount
+            edges {
+              node {
+                artistId
+                medium
+                demandRank
+                annualLotsSold
+                annualValueSoldCents
+                lastAuctionResultDate
+              }
+            }
+          }
```

Comment:
@olerichter00 I am curious about how this might resolve with older versions of Eigen that do not currently do not send the category param when requesting for insights. Maybe I am missing something here.

Reply:
In this case, the client doesn't have to send the category at all. When using `marketPriceInsightsBatchLoader` Metaphysics has already loaded the artwork (with medium and category) and can use the helper `getVortexMedium(artist.medium, artist.category)` to determine which one is the string that needs to be send to Vortex.

Reply:
For example [here](https://github.com/artsy/metaphysics/blob/6dab4f604e7aa345c22339571749ce31ea102802/src/schema/v2/me/myCollection.ts#L291).

Reply:
Ahaaaaa!!!! That was what I was expecting. But I can't figure out at what point in the entire code that happens. I see a default resolution for all insights 👍 , but can't find the logic for a single insight 🙏

Reply:
Yeah, actually we always use this batch loader at the moment for performance reasons...

Reply:
Sorry, there I've refactored the code a lot...